rmw_implementation/src/functions.cpp

```diff
@@ -0,0 +1,513 @@
+// Copyright 2016 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cstddef>
+#include <cstdio>
+#include <cstring>
+
+#include <list>
+#include <string>
+
+#include <fstream>
+#include <sstream>
+
+#include "Poco/SharedLibrary.h"
+
+#include "rmw/rmw.h"
+
+std::string get_env_var(const char * env_var)
```

Comment:
I'd recommend we try to do https://github.com/ros2/rmw/pull/87 (and related pr's) to avoid duplicating this function around.